### PR TITLE
newlib: 3.3.0 -> 4.1.0

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
###### Motivation for this change
Some stuff im working depends on this, thought it would be nice to PR seperately.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126799` at 7bee527f run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>18 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/cf1930f1172748ae53937310afd00296">armTrustedFirmwareAllwinner</a></li>
    <li><a href="https://gist.github.com/3dc4146bdd11452d41be33dfcaaa7472">armTrustedFirmwareQemu</a></li>
    <li><a href="https://gist.github.com/9e7e1114a8e3feea486c99a53a0ab14d">armTrustedFirmwareRK3328</a></li>
    <li><a href="https://gist.github.com/4f46e74a783b6fbacacdb69ab1b99211">armTrustedFirmwareRK3399</a></li>
    <li><a href="https://gist.github.com/360dde7719ea7c6df9719679c3a510b3">armTrustedFirmwareS905</a></li>
    <li><a href="https://gist.github.com/a942d4a28f8a3fb8216f50801007a4ca">ubootBananaPim64</a></li>
    <li><a href="https://gist.github.com/39472a7726f25e62abb22b1b2302454c">ubootNanoPCT4</a></li>
    <li><a href="https://gist.github.com/cb9903a7fe19b9b86faf4e33bcf94d48">ubootOdroidC2</a></li>
    <li><a href="https://gist.github.com/abf38212199bd0fd6774999fffc0caaf">ubootOrangePiZeroPlus2H5</a></li>
    <li><a href="https://gist.github.com/69f468b7cc4cc7caa4a3338d00dfa028">ubootPine64</a></li>
    <li><a href="https://gist.github.com/b8736b9c66cf43f9b2dece9bbf56aa54">ubootPine64LTS</a></li>
    <li><a href="https://gist.github.com/319797532f307522fed33fd2ec7f031d">ubootPinebook</a></li>
    <li><a href="https://gist.github.com/dd66c8588ce67852eeb192a9dbfcfb7e">ubootPinebookPro</a></li>
    <li><a href="https://gist.github.com/55181d12f3f4de2d174b6f473e73855c">ubootROCPCRK3399</a></li>
    <li><a href="https://gist.github.com/5f95f306c762946b48ca17e224d3bbdc">ubootRock64</a></li>
    <li><a href="https://gist.github.com/f8ba5223ab4db53f07df2ced25cd7225">ubootRockPi4</a></li>
    <li><a href="https://gist.github.com/7d489ea3cb20be6c69708191d059cf9e">ubootRockPro64</a></li>
    <li><a href="https://gist.github.com/1cbf172eb50085852a60065653864041">ubootSopine</a></li>
  </ul>
</details>
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>gnuk</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/development/misc/newlib/default.nix [hunk 1]
```diff
@@ -1,12 +1,12 @@
 { stdenv, fetchurl, buildPackages }:
 
-let version = "3.3.0";
-in stdenv.mkDerivation {
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/development/misc/newlib/default.nix:3]: 👍🏼

File: pkgs/development/misc/newlib/default.nix [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
   pname = "newlib";
-  inherit version;
+  version = "4.1.0";
+
   src = fetchurl {
     url = "ftp://sourceware.org/pub/newlib/newlib-${version}.tar.gz";
-    sha256 = "0ricyx792ig2cb2x31b653yb7w7f7mf2111dv5h96lfzmqz9xpaq";
+    sha256 = "0m01sjjyj0ib7bwlcrvmk1qkkgd66zf1dhbw716j490kymrf75pj";
```
Comment (SuperSandro2000) [inline pkgs/development/misc/newlib/default.nix:9]:
```suggestion
    sha256 = "sha256-qfzZ3aVYYn0ZcZgKcLdB13B+eweDebLSBA3HTqdPdv0=";
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 126799` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages marked as broken and skipped:
</summary>
<ul>
<li>betaflight</li>
<li>inav</li>
</ul>
</details>
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>armTrustedFirmwareTools: log was empty</li>
  </ul>
</details>
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>axoloti</li>
<li>fusee-launcher</li>
<li>gnuk</li>
<li>newlib</li>
</ul>
</details>

**Pull request merged**